client/hotrod-client/src/test/java/org/infinispan/client/hotrod/query/aggregation/RemoteClusteredAggregationExpiringTest.java

```diff
@@ -0,0 +1,129 @@
+@TestForIssue(githubKey = "13194")
+public class RemoteClusteredAggregationExpiringTest extends MultiHotRodServersTest {
+
+   private final String indexDirectory = CommonsTestingUtil.tmpDirectory(getClass());
+
+   /*
+   <?xml version="1.0"?>
+   <distributed-cache name="TEST_TASKS" owners="1" mode="SYNC" remote-timeout="17500" statistics="true">
+      <encoding>
+         <key media-type="application/x-protostream"/>
+         <value media-type="application/x-protostream"/>
+      </encoding>
+      <locking concurrency-level="1000" acquire-timeout="15000" striping="false"/>
+      <expiration lifespan="-1"/>
+      <memory storage="HEAP" max-count="1000"/>
+      <persistence passivation="false">
+         <file-store purge="false" read-only="false"/>
+      </persistence>
+      <indexing enabled="true" storage="filesystem">
+         <indexed-entities>
+            <indexed-entity>test.Task</indexed-entity>
+         </indexed-entities>
+      </indexing>
+      <state-transfer timeout="60000"/>
+   </distributed-cache>
+   */
+   @Override
+   protected void createCacheManagers() throws Throwable {
+      Util.recursiveFileRemove(indexDirectory);
+      assertThat(new File(indexDirectory).mkdirs()).isTrue();
+
+      ConfigurationBuilder config = hotRodCacheConfiguration(getDefaultClusteredCacheConfig(CacheMode.DIST_SYNC));
+      config.clustering()
+       .cacheMode(CacheMode.DIST_SYNC)
+            .remoteTimeout(17500)
+            .hash().numOwners(1);
+      config.statistics().enable();
+      config.encoding().key().mediaType(MediaType.APPLICATION_PROTOSTREAM);
+      config.encoding().value().mediaType(MediaType.APPLICATION_PROTOSTREAM);
+      config.locking().concurrencyLevel(1000).lockAcquisitionTimeout(15000);
+      config.expiration().lifespan(-1);
+      config.memory().storage(StorageType.HEAP).maxCount(1000);
+      config.persistence().passivation(false)
+            .addStore(new DummyInMemoryStoreConfigurationBuilder(config.persistence()));
+      config.indexing().enable()
+            .storage(FILESYSTEM).path(indexDirectory)
+            .addIndexedEntity("model.Task");
+      config.clustering().stateTransfer().timeout(60000);
+
+      createHotRodServers(1, config);
+      waitForClusterToForm();
+   }
+
+   @Override
+   protected SerializationContextInitializer contextInitializer() {
+      return Task.TaskSchema.INSTANCE;
+   }
+
+   @AfterClass(alwaysRun = true)
+   @Override
+   protected void destroy() {
+      try {
+         //first stop cache managers, then clear the index
+         super.destroy();
+      } finally {
+         //delete the index otherwise it will mess up the index for next tests
+         Util.recursiveFileRemove(indexDirectory);
+      }
+      super.destroy();
+   }
+
+   @Test
+   public void test() throws Exception {
+      RemoteCache<Integer, Task> remoteCache = clients.get(0).getCache();
+      for (int i=0; i<12; i++) {
```

Comment:
Isn't 2 entries enough? one immortal and another expired and check if the query returns only the immortal entry.

Reply:
I think that if you want to cover a little the case 12 is a number that is quite reasonable.
We want to test more interactions and more expirations.